### PR TITLE
Feat/proposal rubric

### PR DESCRIPTION
**Pull request by motechFR**
### WHAT
copilot:summary

### WHY
Database schema and policies

### CODE REVIEW

File: src/prisma/schema.prisma [hunk 1]
```diff
@@ -1122,6 +1134,41 @@ model ProposalReviewer {
...
+  id          String   @id   @db.Uuid @default(uuid())
+  title       String
+  description String?
+  proposalId  String      @db.Uuid
+  proposal    Proposal @relation(fields: [proposalId], references: [id], onDelete: Cascade)
+  type        ProposalRubricCriteriaType
+  parameters  Json
+  answers     ProposalRubricCriteriaAnswer[]
+
+  @@unique([title, proposalId])
```
Comment (motechFR) [inline src/prisma/schema.prisma:1151]: Make sure criteria don't overlap

File: src/prisma/schema.prisma [hunk 2]
```diff
@@ -1122,6 +1134,41 @@ model ProposalReviewer {
...
+model ProposalRubricCriteriaAnswer {
+  id          String   @id   @db.Uuid @default(uuid())
+  rubricCriteriaId  String   @db.Uuid
+  rubricCriteria  ProposalRubricCriteria   @relation(fields: [rubricCriteriaId], references: [id], onDelete: Cascade)  
+  proposalId  String      @db.Uuid
+  proposal    Proposal @relation(fields: [proposalId], references: [id], onDelete: Cascade)
+  userId      String
+  response    Json
+
+  @@unique([userId, rubricCriteriaId])
```
Comment (motechFR) [inline src/prisma/schema.prisma:1165]: One response per user per criteria

File: src/prisma/schema.prisma [hunk 3]
```diff
@@ -1088,7 +1096,11 @@ model Proposal {
   categoryId             String?            @db.Uuid
   category               ProposalCategory?  @relation(fields: [categoryId], references: [id], onDelete: SetNull)
   snapshotProposalExpiry DateTime?
+  sourceTemplateId       String?            @db.Uuid
```
[the remark this answers is not in the document]
Comment (mattcasey) [reply src/prisma/schema.prisma:1099]: Works for me. if you delete he 'source template', we wouldn't want to delete all the proposals created with it. Right now i just need to know it exists.

[the remark this answers is not in the document]
Comment (mattcasey) [reply src/prisma/schema.prisma:1099]: @motechFR please put this on Page and not Proposal

File: src/prisma/schema.prisma (hunk 1 above)
Comment (mattcasey) [reply src/prisma/schema.prisma:1151]: @motechFR can we actually remove this. there's no technical need to enforce it - i'd rather we put rules like that in the UI instead.

File: src/prisma/schema.prisma [hunk 4]
```diff
@@ -1122,6 +1134,41 @@ model ProposalReviewer {
...
+  parameters  Json
+  answers     ProposalRubricCriteriaAnswer[]
+
+  @@unique([title, proposalId])
+
+  @@index([proposalId])
+}
+
+model ProposalRubricCriteriaAnswer {
+  id          String   @id   @db.Uuid @default(uuid())
```
Comment (mattcasey) [inline src/prisma/schema.prisma:1157]: i don't think answers need an id

File: src/prisma/schema.prisma (hunk 2 above)
Comment (mattcasey) [reply src/prisma/schema.prisma:1165]: this one makes sense 👍

**Review by mattcasey — APPROVED** [review]

**Pull request merged**